transport/event_notifier.cc

```diff
@@ -268,6 +273,10 @@ void cql_server::event_notifier::on_leave_cluster(const gms::inet_address& endpo
 
 void cql_server::event_notifier::on_up(const gms::inet_address& endpoint)
 {
+    if (_endpoints_pending_joined_notification.erase(endpoint)) {
+        on_join_cluster(endpoint);
```

Comment:
Please extract the notification logic from `on_join_cluster()` into it's own function and call that from here. It's confusing if the top-level callback functions start calling each other.

Reply:
Fixed in amended commit: https://github.com/scylladb/scylla/pull/7306/commits/35f19c6d3d2259feb2adb282c65a6da345c055d7